### ISSUE TITLE
During installation error of `pkg-config` installation while `pkg-config` is pre installed on my system.

### ISSUE THREAD

**Issue by Raza403** [sparklemotion/nokogiri#1867]
I am facing `pkg-config` error during installation of nokogiri, although I have pre installed `pkg-config` using both `apt-get` and `gem install`.

Below is complete installation with error.

```ahmed@ahmed-Latitude-E6330:~/test$ bundle install --path vendor/bundle
Fetching gem metadata from https://rubygems.org/.........
Using rake 12.3.1
Using concurrent-ruby 1.0.5
Using i18n 0.9.5
Using minitest 5.11.3
Using thread_safe 0.3.6
Using tzinfo 1.2.5
Using activesupport 5.2.1
Using public_suffix 3.0.2
Using addressable 2.5.2
Using bundler 1.17.3
Using colorator 1.1.0
Using colorize 0.8.1
Using eventmachine 1.2.7
Using http_parser.rb 0.6.0
Using em-websocket 0.5.1
Using ffi 1.9.25
Using ethon 0.11.0
Using forwardable-extended 2.6.0
Using mercenary 0.3.6
Using mini_portile2 2.3.0
Fetching nokogiri 1.8.4
Installing nokogiri 1.8.4 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/home/ahmed/test/vendor/bundle/ruby/2.5.0/gems/nokogiri-1.8.4/ext/nokogiri
/usr/bin/ruby2.5 -I /usr/local/lib/site_ruby/2.5.0 -r
./siteconf20190126-8846-mbo6v6.rb extconf.rb --use-system-libraries
checking if the C compiler accepts ... yes
Building nokogiri using system libraries.
pkg-config could not be used to find libxml-2.0
Please install either `pkg-config` or the pkg-config gem per

    gem install pkg-config -v "~> 1.1"

pkg-config could not be used to find libxslt
Please install either `pkg-config` or the pkg-config gem per

    gem install pkg-config -v "~> 1.1"

pkg-config could not be used to find libexslt
Please install either `pkg-config` or the pkg-config gem per

    gem install pkg-config -v "~> 1.1"

ERROR: cannot discover where libxml2 is located on your system. please make sure
`pkg-config` is installed.
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/bin/$(RUBY_BASE_NAME)2.5
	--help
	--clean
	--use-system-libraries
	--with-zlib-dir
	--without-zlib-dir
	--with-zlib-include
	--without-zlib-include=${zlib-dir}/include
	--with-zlib-lib
	--without-zlib-lib=${zlib-dir}/lib
	--with-xml2-dir
	--without-xml2-dir
	--with-xml2-include
	--without-xml2-include=${xml2-dir}/include
	--with-xml2-lib
	--without-xml2-lib=${xml2-dir}/lib
	--with-libxml-2.0-config
	--without-libxml-2.0-config
	--with-pkg-config
	--without-pkg-config
	--with-xslt-dir
	--without-xslt-dir
	--with-xslt-include
	--without-xslt-include=${xslt-dir}/include
	--with-xslt-lib
	--without-xslt-lib=${xslt-dir}/lib
	--with-libxslt-config
	--without-libxslt-config
	--with-exslt-dir
	--without-exslt-dir
	--with-exslt-include
	--without-exslt-include=${exslt-dir}/include
	--with-exslt-lib
	--without-exslt-lib=${exslt-dir}/lib
	--with-libexslt-config
	--without-libexslt-config

To see why this extension failed to compile, please check the mkmf.log which can
be found here:

/home/ahmed/test/vendor/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/nokogiri-1.8.4/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in
/home/ahmed/test/vendor/bundle/ruby/2.5.0/gems/nokogiri-1.8.4 for inspection.
Results logged to
/home/ahmed/test/vendor/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/nokogiri-1.8.4/gem_make.out

An error occurred while installing nokogiri (1.8.4), and Bundler cannot
continue.
Make sure that `gem install nokogiri -v '1.8.4' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  lightning_sites was resolved to 1.4.0, which depends on
    html-proofer-mailto_awesome was resolved to 0.1.3, which depends on
      html-proofer was resolved to 3.9.2, which depends on
        nokogiri

```
**Screenshot after running `gem list` which shows `pkg-config` installed**
![2](https://user-images.githubusercontent.com/20680336/51783969-f789e880-2163-11e9-8d8b-804be7288655.png)

**Screenshot of commnad `pkg-config --version` also shows `pkg-config` installed**
![3](https://user-images.githubusercontent.com/20680336/51783987-328c1c00-2164-11e9-9ac8-86737c2f2726.png)

**Output of `which pkg-config`**
![4](https://user-images.githubusercontent.com/20680336/51783994-518aae00-2164-11e9-8072-def30c10bd8a.png)

**Comment by flavorjones** [comment]
@Raza403 Thanks for opening this issue.

Because you didn't follow the [github issue template for installation issues](https://github.com/sparklemotion/nokogiri/blob/master/.github/ISSUE_TEMPLATE/installation-difficulties.md), I'm going to ask you the questions from the template to better inform how we approach troubleshooting.

---

**Have you read and followed the advice in the installation tutorial at http://www.nokogiri.org/tutorials/installing_nokogiri.html?**

**What are the complete contents of the `mkmf.log` file generated during the failed installation?**

Please make sure the escape the file contents with triple-backticks.

**What operating system are you using?**

Are you using Linux, MacOS, Windows, a BSD variant, or something else entirely?

If MacOS, please note the version of XCode you have installed.

If Linux, please include the output of `uname -a` and the contents of `/etc/lsb-release`.

If Windows, please note whether you're installing the precompiled gems, or compiling yourself with DevKit. Also please note the version of RubyInstaller you've installed.

**Comment by Raza403** [comment]
@flavorjones Thanks for pointing this out for me, I haven't seen this. I am working on all of the questions asked in the template, will update you on this soon.

**Comment by Raza403** [comment]
Issue solved by updating nokogiri version from 1.8.4 to 1.10.1.
